### DISCUSSION TITLE
Example of filename-fn for deps.edn

### DISCUSSION THREAD

**Discussion by CambodianCoder** [borkdude/quickdoc#25] [Q&A]
I saw examples of using `filename-fn` for a babashka task.

Could you provide an example of using `filename-fn` when used in a deps.edn file?
Use case is to append a sub-directory name in front of the link path that is generated.

Currently attempting the following in deps.edn

```clojure
{...
 :paths ["src" "resources"]
 :aliases {:quickdoc ;; run using clj -M:quickdoc
           {:deps {org.babashka/cli {:mvn/version "0.4.38"}
                   io.github.borkdude/quickdoc
                   {:deps/root "jvm"
                    :git/sha "534c4d09ef052cd5e6896dad9ebe97dc01110264"}}
            :exec-args {:github/repo "https://github.com/{my-org}/{my-repo}"
                        :git/branch "main"
                        :filename-fn (fn [filename]
                                       (clojure.string/replace filename #"^/src" "{project}/src"))}
            :main-opts ["-m" "babashka.cli.exec" "quickdoc.api" "quickdoc"]}}}
```

but this results in the message `Error building classpath. Error reading edn. No dispatch macro for: {path-to-deps.edn}`

Is the filename-fn supposed to be defined in a clojure namespace?

**Comment by borkdude** [comment]
Edn doesn't support regex literals, but you could write `(re-pattern ...)` instead. 
But then we have the problem that isn't isn't interpreted as a function: it's just an s-expr. So I think for this use-case you might be better of writing a script:

quickdoc_script.clj
``` clojure
(ns quickdoc-script
  (:require [quickdoc.api :refer [quickdoc]]))

(defn docs []
  (quickdoc {:filename-fn ...}))
```

and call this from your alias:

``` clojure
:paths ["."]
:exec-fn quickdoc-script/docs
```

**Comment by borkdude** [comment]
@CambodianCoder 

The newest release now has `:filename-add-prefix` and `:filename-remove-prefix` options so the most common things you will do with `:filename-fn` can now be configured in EDN.
